server/src/main/kotlin/com/xebia/functional/xef/server/exceptions/ExceptionsHandler.kt

```diff
@@ -0,0 +1,32 @@
+package com.xebia.functional.xef.server.exceptions
+
+import com.xebia.functional.xef.server.models.exceptions.XefExceptions
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.plugins.statuspages.*
+import io.ktor.server.response.*
+
+fun Application.exceptionsHandler() {
+    install(StatusPages) {
+        exception<Throwable> { call, cause ->
+            if (cause is XefExceptions) {
+                call.manageException(cause)
+            } else {
+                call.respond(HttpStatusCode.InternalServerError, cause.message ?: "Unexpected error")
+            }
+            call.respondText(cause.localizedMessage, status = HttpStatusCode.InternalServerError)
+        }
+        status(HttpStatusCode.NotFound) { call, status ->
+            call.respondText(text = "404: Page Not Found", status = status)
+        }
+    }
+}
+
+suspend fun ApplicationCall.manageException(cause: XefExceptions) {
+    when(cause) {
+        is XefExceptions.ValidationException -> this.respond(HttpStatusCode.BadRequest, cause.message)
+        is XefExceptions.AuthorizationException -> this.respond(HttpStatusCode.Unauthorized)
+        is XefExceptions.OrganizationsException -> this.respond(HttpStatusCode.BadRequest, cause.message)
+        is XefExceptions.UserException -> this.respond(HttpStatusCode.BadRequest, cause.message)
```

Comment:
Perhaps makes sense to have this more granular. To have different types of User exceptions, and return different statuses on each case. The same for the rest of the exceptions 🤔